### PR TITLE
REC-177 Introduce probe that checks top level and component specific statuses

### PR DESCRIPTION
**Pull request by agelostsal**

### CODE REVIEW

**Comment by themiszamani** [comment]
https://argoeu.github.io/argo-monitoring/docs/monitoring/guidelines
please dont forget to add the following parameters 
--help, -h 
--hostname, -H
--verbose, -vvv (for debug reasons)

because there are rules in the monitoring engine. 
Each probe must provide the following arguments:
-h help (--help)
-t timeout (--timeout)
-H hostname (--hostname)

**Review by kaggis — COMMENTED** [review]
specific dependencies for the probe?

File: check_rs.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            if args.component is not None:
+                checks.append(apply_specific_component_check(component=args.component))
+            else:
+                checks.append(apply_status_check)
+            return apply_response_checks(rs_response.json(), *checks)
+
+        LOGGER.error(f"Response status code was {rs_response.status_code}. {str(rs_response.json())}")
+        return critical(f"Response status code was {rs_response.status_code}. {str(rs_response.json())}")
+
+    except requests.exceptions.JSONDecodeError as jsone:
```
Comment (kaggis) [inline check_rs.py]: Is this supported only in latest versions (Jun 2022 onwards)?

Comment (agelostsal) [reply check_rs.py]: it is introduced in python 3.6 which is the version that we package our rpms, so we should be safe to use it.

**Comment by themiszamani** [comment]
+1

**Pull request merged**